spec/controllers/admin/reports/registered_adviser_controller_spec.rb

```diff
@@ -0,0 +1,35 @@
+RSpec.describe Admin::Reports::RegisteredAdviserController, type: :controller do
+  describe '#filename' do
+    before(:each) do
+      Timecop.freeze(date)
+    end
+
+    after(:each) do
+      Timecop.return
+    end
+
+    context 'yesterday' do
```

Comment:
I disagree — the problem is that as a human "yesterday" has a very specific meaning. By setting another forced date in the tests, you have done nothing to remove the confusion that "yesterday" in this context is not "yesterday".

IMO you only need two tests, with very different arbitrary dates (one to prove the filename contains the date, the second to prove that it wasn't hardcoded to that date but is actually dependent on the system date at the time code is executed). Doing yesterday/today/tomorrow is cute but ultimately confusing once enough time has passed.